programs/autocrat_v0/src/lib.rs

```diff
@@ -381,18 +388,44 @@ pub mod autocrat_v0 {
         Ok(())
     }
 
-    pub fn set_last_proposal_slot(ctx: Context<Auth>, last_proposal_slot: u64) -> Result<()> {
+    pub fn set_base_burn_lamports(ctx: Context<Auth>, base_burn_lamports: u64) -> Result<()> {
         let dao = &mut ctx.accounts.dao;
 
-        dao.last_proposal_slot = last_proposal_slot;
+        dao.base_burn_lamports = base_burn_lamports;
 
         Ok(())
     }
 
-    pub fn set_base_burn_lamports(ctx: Context<Auth>, base_burn_lamports: u64) -> Result<()> {
+    pub fn set_burn_decay_per_slot_lamports(
+        ctx: Context<Auth>,
+        burn_decay_per_slot_lamports: u64,
+    ) -> Result<()> {
         let dao = &mut ctx.accounts.dao;
 
-        dao.base_burn_lamports = base_burn_lamports;
+        dao.burn_decay_per_slot_lamports = burn_decay_per_slot_lamports;
+
+        Ok(())
+    }
+
+    pub fn set_slots_per_proposal(ctx: Context<Auth>, slots_per_proposal: u64) -> Result<()> {
+        let dao = &mut ctx.accounts.dao;
+
+        dao.slots_per_proposal = slots_per_proposal;
+
+        Ok(())
+    }
+
+    pub fn set_market_taker_fee(ctx: Context<Auth>, market_taker_fee: i64) -> Result<()> {
+        let dao = &mut ctx.accounts.dao;
+
+        dao.market_taker_fee = market_taker_fee;
+
+        Ok(())
```

Comment:
would the fees accumulate in the conditional token markets? who is the owner to withdraw form there? the proposal initializer or the dao? if its the dao, would this program be able to if immutable?

Reply:
It needs to be paid out to the makers by makerVolume as a % of total collected fees. 

Not sure if that could be called on market close or how resolution occurs.

Maybe it could be part of the redeem function as it'd be associated with the account?

State gets a bit muddy if people can still trade the markets after proposal is completed, how would you flag an account as having redeemed, but then able to redeem more if the market still trades?

Reply:
Right now it just goes to the treasury, we could add the other functionality as a future version of autocrat